src/Task.ts

```diff
@@ -104,14 +104,24 @@ export class TaskRegularExpressions {
     public static readonly doneDateRegex = /✅ *(\d{4}-\d{2}-\d{2})$/u;
     public static readonly recurrenceRegex = /🔁 ?([a-zA-Z0-9, !]+)$/iu;
 
-    // Regex to match all hash tags, basically hash followed by anything but the characters in the negation.
+    // Create a regex to match all hash tags, basically hash followed by anything but the characters in the negation.
     // To ensure URLs are not caught it is looking of beginning of string tag and any
     // tag that has a space in front of it. Any # that has a character in front
     // of it will be ignored.
     // EXAMPLE:
     // description: '#dog #car http://www/ddd#ere #house'
     // matches: #dog, #car, #house
-    public static readonly hashTags = /(^|\s)#[^ !@#$%^&*(),.?":{}|<>]*/g;
+    public static readonly hashTagsValidChars = /[^ !@#$%^&*()[\],.?":;{}|<>]/;
```

Comment:
So did some poking in Obsidian a while back around tags to see how they handled it. 

This is the process they follow.

Iterate through the string until they hit the hashtag '#'
Once found then they slice the string starting at position 1 (removes the '#')
Then do a replace on the remaining string to remove any '\' characters. (replace(/\\./g, ""))
They then run the follow regex /^(?:[^\u2000-\u206F\u2E00-\u2E7F'!"#$%&()*+,.:;<=>?@^`{|}~\[\]\\\s])+/gm
If the regex exists and it matches a /[^0-9]/.test(firstresultfromlastregex) so it has content and is not just numbers.
We now have a tag that can be used.

To implement this in tasks it needs to for a for loop through the task line and then run the match. 

I was looking at moving away from the regex based processing to token based processing (tokenizer) and a for loop before I got to busy as it would make some of the processing easier and additional tokens could be added. This is like how obsidian does it when processing the markdown to render.

Reply:
Thank you. That's interesting.

I was thinking that replicating Obsidian's code would be a bit of a maintenance headache - and it would be safer to use Obsidian's date - as described in [this comment](https://github.com/obsidian-tasks-group/obsidian-tasks/issues/929#issuecomment-1257069606).

Then if they change their implementation, we automatically get the new behaviour.